### PR TITLE
[7.x] Adding limitOnChar Str helper function

### PR DESCRIPTION
**Pull request by intraordinaire**
Adding a util method in the `\Illuminate\Support\Str` helper class.

`Str::limitOnChar` makes a limit on the last occurrence of space (or another given char).
It can be very useful to make short descriptions, or sending mail, with truncated texts without cutting sentences in a middle of a word.

Example:
```php
use Illuminate\Support\Str;

Str::limitOnChar('The PHP framework for web artisans.', 20); // The PHP framework...
Str::limit('The PHP framework for web artisans.', 20); // The PHP framework fo...
```

If the givent character (default is space) is not found, the behavior is the same as the `Str::limit` method.

```php
use Illuminate\Support\Str;

Str::limitOnChar('ThePHPFrameworkForWebArtisans', 6, ''); //ThePHP
```

### CODE REVIEW

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -340,6 +340,36 @@ public static function limit($value, $limit = 100, $end = '...')
...
+    /**
+     * Limit the number of characters in a string on the last space before the limit..
+     *
+     * @param  string  $value
+     * @param  int  $limit
+     * @param  string  $end
+     * @param  string  $spaceChar
+     * @return string
+     */
+    public static function limitOnSpace($value, $limit = 100, $end = '...', $spaceChar = ' ')
```
Comment (seriquynh) [inline src/Illuminate/Support/Str.php:352]: You name this function "limitOnSpace", so what is the meaning of $spaceChar? When someone passes "." character, it has to be "limitOnDot", not "limitOnSpace" anymore. You can consider "limitByChar" or something similar.

Comment (intraordinaire) [reply src/Illuminate/Support/Str.php:352]: Sure, good point.
Did it for space, and forgot to rename it after adding the extra param.

**Comment by mfn** [comment]
This looks very specific to me, not sure if this is a good fit for core

It might seem logical to break before a character, but there are also other views that it should "break on word boundaries" for example. This is what I would expect from such a feature. A word boundary means it can break about `long-concatenated-words-like-this-too`. But this wouldn't work at all with this method, as its a different approach, so 🤷‍♀️

**Comment by intraordinaire** [comment]
@mfn Yep I know it's a bit specific and I totally understand you're point.
But actually, I use this kind of logic on almost all my projects, and I guess I'm not alone.
I will just keep it to my helpers if not suitable for core.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by khanalpride** [comment]
@intraordinaire I think you can make use of Macroable in this case. 
There are many functions I keep as my helper files and Macros to help me out through different projects. So I hope its always good to have helpers files or Macros list that can be used throughout different projects. When that files get larger, you can separate them into many files and release them as helper package.

**Pull request closed without merging**